src/spyglass/spikesorting/v1/utils.py

```diff
@@ -62,29 +71,22 @@ def get_spiking_sorting_v1_merge_ids(restriction: dict):
     ]
     # list of sorting ids for each recording
     sorting_restriction = restriction.copy()
-    del sorting_restriction["interval_list_name"]
+    _ = sorting_restriction.pop("interval_list_name", None)
     sorting_id_list = []
     for r_id, a_id in zip(recording_id_list, artifact_id_list):
+        rec_dict = {"recording_id": str(r_id), "interval_list_name": str(a_id)}
         # if sorted with artifact detection
-        if (
-            SpikeSortingSelection()
-            & sorting_restriction
-            & {"recording_id": r_id, "interval_list_name": a_id}
-        ):
+        if SpikeSortingSelection() & sorting_restriction & rec_dict:
```

Comment:
Is there any efficiency difference between:
```
SpikeSortingSelection() & sorting_restriction & rec_dict
```
and:
```
SpikeSortingSelection & sorting_restriction & rec_dict
```
?

Or will that not work in this case?

Reply:
When restricting, there's no reason to initialize a table with `()` - dj assumes you want the initialized version if you're restricting